tests/unit/bond_test.py

```diff
@@ -114,6 +116,117 @@ def test_generate_access_token(self):
         self.assertEqual(self.fake_access_token, access_token)
         self.assertEqual(datetime.fromtimestamp(self.expires_at_epoch), expires_at)
 
+    def test_get_access_token_from_cache(self):
+        data = {"context": {"user": {"name": self.name}}, 'iat': self.issued_at_epoch}
+        encoded_jwt = jwt.encode(data, 'secret', 'HS256')
+        expires_at_initial = datetime.now().timestamp() + 1
+        
+        fake_token_defaults = {
+            FenceKeys.ACCESS_TOKEN: self.fake_access_token,
+            FenceKeys.ID_TOKEN: encoded_jwt,
+            }
+
+        mock_oauth_adapter = OauthAdapter("foo", "bar", "baz", "qux")
+        mock_oauth_adapter.refresh_access_token = MagicMock(
+            return_value={**fake_token_defaults, **{FenceKeys.EXPIRES_AT: expires_at_initial}}
+        )
+
+        fence_api = self._mock_fence_api(json.dumps({"private_key_id": "asfasdfasdf"}))
+        cache_api = FakeCacheApi()
+        bond = Bond(mock_oauth_adapter,
+                    fence_api,
+                    cache_api,
+                    self.refresh_token_store,
+                    FenceTokenVendingMachine(fence_api, cache_api, self.refresh_token_store,
+                                             mock_oauth_adapter, provider_name,
+                                             FakeFenceTokenStorage()),
+                    provider_name,
+                    "/context/user/name",
+                    {})
+        
+        token = str(uuid.uuid4())
+        self.refresh_token_store.save(user_id=self.user_id,
+                                      refresh_token_str=token,
+                                      issued_at=datetime.fromtimestamp(self.issued_at_epoch),
+                                      username=self.name,
+                                      provider_name=provider_name)
+
+        # verify that the new `expired_at` value is returned with `get_access_token`,
```

Comment:
Nit:
```suggestion
        # verify that the new `expires_at` value is returned with `get_access_token`,
```
Also on lines 212 and 223.